doc/reverse_engineering/switching_villager_tasks.md

```diff
@@ -0,0 +1,19 @@
+# Switching tasks
+
+[Source](https://www.youtube.com/watch?v=FK__AnTL400)
+
+## Losing carried resources
+
+In the original Age of Kings game, villagers would lose any resources they carried when they switched roles, e.g. from "farmer" to "builder". This was likely caused by AoE "replacing" a villager when they are sent to perform another task. In Age of Conquerors this was slightly changed so that builders could carry resources and would only lose them if they would not switch back to their original role.
+
+AoE 2 : HD Edition further removed some instances where a player would lose resources. Now resources are only lost if the villager gets assigned to a resource that is different from what they are carrying. Food that was gathered in the role of a "shepard" will therefore carry over if the new role is "hunter", "gatherer" or "farmer".
+
+Looking at the first entry from **Weird AoE2 Quirks** it can be assumed that the HD Edition probably uses a dirty workaround to prevent the player from losing resources. The food is most likely put into the newly assigned sheep/boar/deer and then immediatly "collected" again.
+
+## Automatic Drop-off
+
+In addition to the manual method, workers will automatically drop off **any** resource they are carrying to the player's stockpile, if they constructed a gathering site. The type of gathering site does not matter. For example, villagers will drop off food at a mining camp, if they helped finishing the building. The same happens when they build a farm.
```

Comment:
Yes, I didn't know that either. Spirit tested it for AoE2:HD but it's also in AoC 1.0c. In AoK builders couldn't carry resources.

I modified the text because one could misunderstand this as "When they construct it they can now use it as a drop off point for any resource indefinitely". This only happens *after* they finished the site. Spirit explains this [here](https://youtu.be/FK__AnTL400?t=2m51s)